### PR TITLE
#1-4 Task. sergey.vialov 

### PR DESCRIPTION
**Pull request by yippiekay**
https://travis-ci.com/github/yippiekay/js-assignments/builds/175078222

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -238,7 +249,11 @@ function encodeToRot13(str) {
  *   isString(new String('test')) => true
  */
 function isString(value) {
-    throw new Error('Not implemented');
+    
+    if(typeof value === 'string' || value instanceof String)
```
Comment (AlexMokin) [inline task/01-strings-tasks.js:253]: just return bolean expression - `typeof value === 'string' || value instanceof String`

File: task/02-numbers-tasks.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ function getAverage(value1, value2) {
  *   (-5,0) (10,-10) => 18.027756377319946
  */
 function getDistanceBetweenPoints(x1, y1, x2, y2) {
-    throw new Error('Not implemented');
+    return Math.sqrt((x2 - x1) ** 2  + (y2 - y1) ** 2);
```
Comment (AlexMokin) [inline task/02-numbers-tasks.js:76]: try to use math.hypot

File: task/02-numbers-tasks.js [hunk 3]
```diff
@@ -160,7 +162,7 @@ function parseNumberFromString(value) {
  *   1,2,3   => 3.741657386773941
  */
 function getParallelipidedDiagonal(a,b,c) {
-    throw new Error('Not implemented');
+    return Math.sqrt(a ** 2 + b ** 2 + c ** 2);
```
Comment (AlexMokin) [inline task/02-numbers-tasks.js:165]: try to use math.hypot

File: task/03-date-tasks.js [hunk 4]
```diff
@@ -56,10 +56,19 @@ function parseDataFromIso8601(value) {
  *    Date(2015,1,1)    => false
  */
 function isLeapYear(date) {
-   throw new Error('Not implemented');
+  
+   let dateObj = new Date(date);
+   let year = dateObj.getFullYear();
+   if(year % 4)
+   return false;
```
Comment (AlexMokin) [inline task/03-date-tasks.js:63]: use `{}`. also there should be an indentation in front of return

File: task/04-arrays-tasks.js [hunk 5]
```diff
@@ -249,7 +252,8 @@ function getMovingSum(arr) {
  * [ "a" ] => []
  */
 function getSecondItems(arr) {
-   throw new Error('Not implemented');
+   let value = true;
+   return arr.filter(() => value = !value);
```
Comment (AlexMokin) [inline task/04-arrays-tasks.js:256]: see https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

filter also has an index argument, try to do it without `value` variable

File: task/04-arrays-tasks.js [hunk 6]
```diff
@@ -536,7 +576,14 @@ function selectMany(arr, childrenSelector) {
  *   [[[ 1, 2, 3]]], [ 0, 0, 1 ]      => 2        (arr[0][0][1])
  */
 function getElementByIndexes(arr, indexes) {
```
Comment (AlexMokin) [inline task/04-arrays-tasks.js:578]: try to do it with `reduce ` function for any number of indexes

**Comment by AlexMokin** [comment]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by AlexMokin** [comment]
looks good. Pr is closed

**Comment by yippiekay** [comment]
Thanks a lot!

**Pull request closed without merging**